applications/MeshMovingApplication/python_scripts/ale_fluid_solver.py

```diff
@@ -0,0 +1,227 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing the Kratos Library
+import KratosMultiphysics
+KratosMultiphysics.CheckRegisteredApplications("MeshMovingApplication")
+import KratosMultiphysics.MeshMovingApplication
+
+# Other imports
+from python_solver import PythonSolver
+import python_solvers_wrapper_mesh_motion
+
+
+def CreateSolver(model, solver_settings, parallelism):
+    return ALEFluidSolver(model, solver_settings, parallelism)
+
+
+class ALEFluidSolver(PythonSolver):
+    def __init__(self, model, solver_settings, parallelism):
+        super(ALEFluidSolver, self).__init__(model, solver_settings)
+        mesh_motion_solver_settings = solver_settings["ale_settings"].Clone()
+        solver_settings.RemoveValue("ale_settings")
+
+        fluid_model_part_name = solver_settings["model_part_name"].GetString()
+        if not self.model.HasModelPart(fluid_model_part_name):
+            fluid_mesh_model_part = KratosMultiphysics.ModelPart(fluid_model_part_name)
+            self.model.AddModelPart(fluid_mesh_model_part)
+
+        ## Creating the fluid solver
+        self.fluid_solver = self._CreateFluidSolver(solver_settings, parallelism)
+
+        ## Creating the mesh-motion solver
+        if not mesh_motion_solver_settings.Has("echo_level"):
+            mesh_motion_solver_settings.AddValue("echo_level", solver_settings["echo_level"])
+
+        if mesh_motion_solver_settings.Has("model_part_name"):
+            if not fluid_model_part_name == mesh_motion_solver_settings["model_part_name"].GetString():
+                raise Exception('Fluid- and Mesh-Solver have to use the same "model_part_name"!')
+        else:
+            mesh_motion_solver_settings.AddValue("model_part_name", solver_settings["model_part_name"])
+
+        domain_size = solver_settings["domain_size"].GetInt()
+        if mesh_motion_solver_settings.Has("domain_size"):
+            mesh_motion_domain_size = mesh_motion_solver_settings["domain_size"].GetInt()
+            if not domain_size == mesh_motion_domain_size:
+                raise Exception('Fluid- and Mesh-Solver have to use the same "domain_size"!')
+        else:
+            mesh_motion_solver_settings.AddValue("domain_size", solver_settings["domain_size"])
+
+        self.ale_interface_parts_params = None
```

Comment:
why not raising error when `ale_settings` is not present in the parameters? I say it because ale fluid solver with out this is a bit weird. I mean ale fluid solver is used when boundary is moving and when it is moving it should be in the list or?

Reply:
Good point, will change it